### PR TITLE
chore: deploy data-mgt and redis on k8s

### PR DESCRIPTION
**Pull request by danielogen**
# Deploy data-mgt & redis on k8s

**_WHAT DOES THIS PR DO?_**
This PR adds configurations for the deployment of `data-mgt` and `redis` on k8s 
**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
N/A
**_WHAT IS THE LINK TO THE PR BRANCH_**
N/A
**_HOW DO I TEST OUT THIS PR?_**
Locally: `npm run stage-mac` or `npm run stage-pc`. Make sure redis is running 
Docker (stage). `docker-compose up`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
All (endpoints)[https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit#gid=757070984] of data management microservice.
Example endpoint: `http://34.78.78.202:31001/api/v1/data/feeds/transform/recent?channel=689750`
**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
N/A
**_ARE THERE ANY RELATED PRs?_**
N/A

### CODE REVIEW

**Review by Baalmart — APPROVED** [review]
LGTM, thanks.

File: src/data-mgt/node/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: "2"
```
Comment (Baalmart) [inline src/data-mgt/node/docker-compose.yaml:1]: Will the docker compose file also have prod, stage and etc corresponding files?

File: k8s/prod-redis-server.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+apiVersion: apps/v1
+kind: Deployment
```
Comment (Baalmart) [inline k8s/prod-redis-server.yaml:2]: A quick one, what is the main difference between `prod` and `stage` for Redis?
Is it the NodePort? Or the image url/version?

File: src/data-mgt/node/config/constants.js [hunk 3]
```diff
@@ -2,16 +2,22 @@ const devConfig = {
...
   JWT_SECRET: process.env.JWT_SECRET,
   DB_NAME: process.env.MONGO_STAGE,
+  REDIS_SERVER: process.env.REDIS_SERVER,
+  REDIS_PORT: process.env.REDIS_PORT,
 };
 const prodConfig = {
   MONGO_URI: process.env.MONGO_GCE_URI,
   JWT_SECRET: process.env.JWT_SECRET,
   DB_NAME: process.env.MONGO_PROD,
+  REDIS_SERVER: process.env.REDIS_SERVER,
```
Comment (Baalmart) [inline src/data-mgt/node/config/constants.js:19]: The updated .env can be shared in our private Slack channel. Good idea? Happy to hear your thoughts.

**Review by Baalmart — COMMENTED** [review]
Already merged, just a few "after thoughts"......

File: src/data-mgt/node/config/constants.js (hunk 3 above)
Comment (danielogen) [reply src/data-mgt/node/config/constants.js:19]: Alright... sounds good. Let me share it in the `dev-kampala` private channel

File: k8s/prod-redis-server.yaml (hunk 2 above)
Comment (danielogen) [reply k8s/prod-redis-server.yaml:2]: For now, it's just the `k8s namespace`. Both staging and production are being served using the same redis image

**Pull request merged**